fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImplSpec.groovy

```diff
@@ -0,0 +1,53 @@
+package com.yahoo.bard.webservice.druid.client.impl
+
+import com.yahoo.bard.webservice.druid.client.DruidClientConfigHelper
+import com.yahoo.bard.webservice.druid.model.query.QueryContext
+import com.yahoo.bard.webservice.druid.model.query.WeightEvaluationQuery
+
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import io.netty.handler.codec.http.HttpHeaders
+import spock.lang.Specification
+
+import java.util.function.Supplier
+
+class AsyncDruidWebServiceImplSpec extends Specification {
+    def "Ensure that headersToAppend are added"() {
+        setup:
+        WeightEvaluationQuery weightEvaluationQuery = Mock(WeightEvaluationQuery)
+        QueryContext queryContext = Mock(QueryContext)
+        weightEvaluationQuery.getContext() >> { queryContext }
+        queryContext.numberOfQueries() >> { 1 }
+        queryContext.getSequenceNumber() >> { 1 }
+
+        when:
+        Map<String, String> expectedHeaders = new HashMap<>()
+        expectedHeaders.put("k1", "v1")
+        expectedHeaders.put("k2", "v2")
+        Supplier<Map<String, String>> supplier = new Supplier<Map<String, String>>() {
+            @Override
+            Map<String, String> get() {
+                return expectedHeaders
+            }
+        }
+
+        AsyncDruidWebServiceImplWrapper webServiceImplWrapper = new AsyncDruidWebServiceImplWrapper(
+                DruidClientConfigHelper.getNonUiServiceConfig(),
+                new ObjectMapper(),
+                supplier
+        )
```

Comment:
All of this stuff from here up that's in the `when` block should go up in the `setup` block, since it's still setting up the scenario.

If needed, you can use `and:` blocks (with an optional comment: `and: "a set of expected headers to add"`) to break up any block.